### PR TITLE
fix: Fix calendar range end

### LINKED ISSUE #17250: Selecting the same day in "custom fixed date range" will show no events in event explorer
## Bug description

When selecting the same start and end date as the "custom fixed date range", the event explorer will show not events, because the query looks for events which are:
- greater than `toDateTime64('2023-08-23 00:00:00.000000', 6, 'UTC')`
- less than `toDateTime64('2023-08-23 00:00:00.000000', 6, 'UTC')`

I would expect to see all events from the `2023-08-23`

## How to reproduce

1. Open event explorer
2. Select "custom fixed date range"
3. Select same start and ending day

A solution here would be to not allow the same day to be selected.

## Additional context

related ticket: https://posthoghelp.zendesk.com/agent/tickets/5458
clickhouse query: [here](http://metabase-prod-us/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJTRUxFQ1QgdHVwbGUoZXZlbnRzLnV1aWQsIGV2ZW50cy5ldmVudCwgZXZlbnRzLnByb3BlcnRpZXMsIHRvVGltZVpvbmUoZXZlbnRzLnRpbWVzdGFtcCwgJ1VUQycpLCBldmVudHMudGVhbV9pZCwgZXZlbnRzLmRpc3RpbmN0X2lkLCBldmVudHMuZWxlbWVudHNfY2hhaW4sIHRvVGltZVpvbmUoZXZlbnRzLmNyZWF0ZWRfYXQsICdVVEMnKSksXG4gICAgICAgZXZlbnRzLmV2ZW50LFxuICAgICAgIGV2ZW50cy5kaXN0aW5jdF9pZCxcbiAgICAgICByZXBsYWNlUmVnZXhwQWxsKG51bGxJZihudWxsSWYoSlNPTkV4dHJhY3RSYXcoZXZlbnRzLnByb3BlcnRpZXMsICdhcnRpY2xlSUQnKSwgJycpLCAnbnVsbCcpLCAnXlwifFwiJCcsICcnKSxcbiAgICAgICByZXBsYWNlUmVnZXhwQWxsKG51bGxJZihudWxsSWYoSlNPTkV4dHJhY3RSYXcoZXZlbnRzLnByb3BlcnRpZXMsICdrYklEJyksICcnKSwgJ251bGwnKSwgJ15cInxcIiQnLCAnJyksXG4gICAgICAgbnVsbElmKG51bGxJZihldmVudHMuYG1hdF8kbGliYCwgJycpLCAnbnVsbCcpLFxuICAgICAgIHRvVGltZVpvbmUoZXZlbnRzLnRpbWVzdGFtcCwgJ1VUQycpXG5GUk9NIGV2ZW50c1xuV0hFUkUgYW5kKGVxdWFscyhldmVudHMudGVhbV9pZCwgMjAyOCksIGlmTnVsbChsZXNzKHRvVGltZVpvbmUoZXZlbnRzLnRpbWVzdGFtcCwgJ1VUQycpLCB0b0RhdGVUaW1lNjQoJzIwMjMtMDgtMjMgMDA6MDA6MDAuMDAwMDAwJywgNiwgJ1VUQycpKSwgMCksIGlmTnVsbChncmVhdGVyKHRvVGltZVpvbmUoZXZlbnRzLnRpbWVzdGFtcCwgJ1VUQycpLCB0b0RhdGVUaW1lNjQoJzIwMjMtMDgtMjMgMDA6MDA6MDAuMDAwMDAwJywgNiwgJ1VUQycpKSwgMCkpXG5PUkRFUiBCWSB0b1RpbWVab25lKGV2ZW50cy50aW1lc3RhbXAsICdVVEMnKSBERVNDXG5MSU1JVCAxMDFcbk9GRlNFVCAwIFNFVFRJTkdTIHJlYWRvbmx5PTIsXG4gICAgICAgICAgICAgICAgICBtYXhfZXhlY3V0aW9uX3RpbWU9NjAsXG4gICAgICAgICAgICAgICAgICBhbGxvd19leHBlcmltZW50YWxfb2JqZWN0X3R5cGU9VHJ1ZSIsInRlbXBsYXRlLXRhZ3MiOnt9fSwiZGF0YWJhc2UiOjM2fSwiZGlzcGxheSI6InRhYmxlIiwicGFyYW1ldGVycyI6W10sInZpc3VhbGl6YXRpb25fc2V0dGluZ3MiOnt9fQ==)

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

The date range picker picked full days (like ISO `2023-12-01` to `2023-12-02`) as range start and end, which the queries treat as from and to, or some as after and before. All queries will treat the end as meaning `00:00:00` in time however, so not show the day's data as intended.

Fixes  #17250

## Changes

Use the end of the day (= `23:59:59`) as the range end for the calendar range picker.

The reasoning is that we probably don't want to adjust all queries to treat just a day that is given as then end as the full day. Rather this change of the date picker seems pretty straightforward.

Additionally there was a glaring bug in making the start and end show up when loading the picker with state.

## How did you test this code?

- pick specific dates, look at query
- pick a single date, look that data for full day appears
- checked event table and trends page and others

### CODE REVIEW

File: frontend/src/lib/components/DateFilter/DateFilter.tsx [hunk 1]
```diff
@@ -97,7 +97,7 @@ export function DateFilter({
     const popoverOverlay =
         view === DateFilterView.FixedRange ? (
             <LemonCalendarRange
-                value={[rangeDateTo ?? dayjs(), rangeDateTo ?? dayjs()]}
```
Comment (webjunkie) [inline frontend/src/lib/components/DateFilter/DateFilter.tsx:100]: 🐛  this set the date as from `to` to `to`

File: posthog/hogql_queries/utils/query_date_range.py [hunk 2]
```diff
@@ -250,22 +249,6 @@ def to_placeholders(self) -> Dict[str, ast.Expr]:
...
         }
 
-    def to_properties(self, field: Optional[List[str]] = None) -> List[ast.Expr]:
-        if not field:
-            field = ["timestamp"]
-        return [
-            ast.CompareOperation(
-                left=ast.Field(chain=field),
-                op=CompareOperationOp.LtEq,
-                right=self.date_to_as_hogql(),
```
Comment (webjunkie) [inline posthog/hogql_queries/utils/query_date_range.py:260]: This should be `date_to` here as well, but the whole method is unused. Hence removed.

**Review by webjunkie — COMMENTED** [review]
Notes

**Review by mariusandra — APPROVED** [review]
Looks good and makes sense to me! I don't know what if anything will regress because of it. Couldn't find anything at a small search through the codebase, so let's 🚢 and 👀.

**Pull request merged**